jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -209,6 +213,45 @@ YUI.add('user-profile', function() {
     },
 
     /**
+      Fetches the model name from the modelName ref and then creates a model
+      then switches to it.
+      @method createAndSwitch
+      @param {Object} e The event handler from a form submission.
+    */
+    createAndSwitch: function(e) {
```

Comment:
Yeah I was thinking the same but hopefully all of this will go away soon. (famous last words ;) )